.github/workflows/test.yml

```diff
@@ -4,14 +4,14 @@ jobs:
   test:
     strategy:
       matrix:
-        go-versions: [1.12.x, 1.13.x, 1.14.x]
+        go: ["1.13.x", "1.14.x", "1.15.x"]
         platform: [ubuntu-latest]
     runs-on: ${{ matrix.platform }}
     steps:
-    - name: Install Go
-      uses: actions/setup-go@v1
+    - name: Install Go ${{ matrix.go }}
+      uses: actions/setup-go@v2
       with:
-        go-version: ${{ matrix.go-version }}
+        go-version: ${{ matrix.go }}
```

Comment:
Why change this line?

Reply:
i think it is`go-versions` in line 7 and `go-version` here, so it didn't work.

Reply:
Ah, yes, so here's why I renamed (and why it was somewhat easy to overlook the problem);

Originally, the matrix looked like (simplified);

```yaml
strategy:
  matrix:
    go-versions: [1.12.x, 1.13.x, 1.14.x]
steps:
- name: Install Go
  uses: actions/setup-go@v1
  with:
    go-version: ${{ matrix.go-version }}
```

I haven't done a lot with GH Actions yet, so when reading the above, I made the incorrect assumption that was how GH actions handled matrix values;

- `matrix -> go-versions` being plural (because it's a list), and
- `with -> go-version` singular, because it gets assigned a value from the matrix.

Clearly, I was wrong there (and the `with:` value must have the same name as the value in the `matrix:`).

When looking for the problem, I went to the GitHub repo of the action; https://github.com/actions/setup-go/blob/v2.1.3/README.md#usage. Comparing those examples with what we had;

- Perhaps it was because we didn't explicitly add quotes around the values in the `1.12.x` (recalling the dreaded https://github.com/docker/compose/issues/103 issue when specifying port-mappings in compose)
- Perhaps it didn't support the `.x` suffixes? (planning on opening a ticket or PR in that repository to add a section explaining what formats are accepted)
- Perhaps it was a bug in v1, fixed in v2?
- Perhaps it was the name (`go-versions` versus `go`)?

And that's when I found the difference (plural versus singular).

Reply:
@AkihiroSuda I can rename back to `go-version` if you prefer; I used `go` (as in the example) because I thought it was less likely for someone to rename `go` to `gos` (plural), and re-introduce the bug, but I'm not really attached to the name